rideshare.rb

```diff
@@ -0,0 +1,126 @@
+drivers = [
+  {driver_id: "DR0001",
+   trips: [
+    {date: "3rd Feb 2016",
+     cost: 5,
+     rider_id: "RD0003",
+     rating: 3},
+    {date: "3rd Feb 2016",
+     cost: 30,
+     rider_id: "RD0015",
+     rating: 4},
+    {date: "5th Feb 2016",
+     cost: 45,
+     rider_id: "RD0003",
+     rating: 2},
+  ]},
+  {driver_id: "DR0002",
+   trips: [
+    {date: "3rd Feb 2016",
+     cost: 25,
+     rider_id: "RD0073",
+     rating: 5},
+    {date: "4th Feb 2016",
+     cost: 15,
+     rider_id: "RD0013",
+     rating: 1},
+    {date: "5th Feb 2016",
+     cost: 35,
+     rider_id: "RD0066",
+     rating: 3},
+  ]},
+  {driver_id: "DR0003",
+   trips: [
+    {date: "4th Feb 2016",
+     cost: 5,
+     rider_id: "RD0066",
+     rating: 5},
+    {date: "5th Feb 2016",
+     cost: 50,
+     rider_id: "RD0003",
+     rating: 2},
+  ]},
+  {driver_id: "DR0004",
+   trips: [
+    {date: "3rd Feb 2016",
+     cost: 5,
+     rider_id: "RD0022",
+     rating: 5},
+    {date: "4th Feb 2016",
+     cost: 10,
+     rider_id: "RD0022",
+     rating: 4},
+    {date: "5th Feb 2016",
+     cost: 20,
+     rider_id: "RD0073",
+     rating: 5},
+  ]},
+]
+
+def print_summary(drivers, relevant_array, units_string)
+  ids = drivers.map do |driver|
+    driver[:driver_id]
+  end
+  return ids.zip(relevant_array).map do |pair|
+           puts "#{pair[0]}: #{pair[1]} #{units_string}"
+         end
```

Comment:
I really like how general this method is.  

One thing to note though its that it relies on none of your other methods re-ordering `relevant_array` vs `drivers`.  This works fine here though if you use a solution like this in the future you will want to be very careful about enforcing an ordering in the rest of your code.